pedersen/src/product_protocol.rs

```diff
@@ -0,0 +1,557 @@
+        cx: &Self::Affine,
+        cy: &Self::Affine,
+        cxy: &Self::Affine,
+    );
+}
+
+impl<P: PedersenConfig> ProductProofTranscriptable for ProductProof<P> {
+    type Affine = sw::Affine<P>;
+    fn add_to_transcript(
+        &self,
+        transcript: &mut Transcript,
+        cx: &Self::Affine,
+        cy: &Self::Affine,
+        cxy: &Self::Affine,
+    )
+    {
+        ProductProof::make_transcript(
+            transcript, 
+            cx, 
+            cy, 
+            cxy,
+            &self.alpha,
+            &self.beta,
+            &self.delta,
+        );
+    }
+}
+
+impl<P: PedersenConfig> ProductProofTranscriptable for ProductProofIntermediate<P> {
+    type Affine = sw::Affine<P>;
+    fn add_to_transcript(
+        &self,
+        transcript: &mut Transcript,
+        cx: &Self::Affine,
+        cy: &Self::Affine,
+        cxy: &Self::Affine,
+    ) {
+        ProductProof::make_transcript(
+            transcript,
+            cx, 
+            cy,
+            cxy,
+            &self.alpha,
+            &self.beta,
+            &self.delta,
+        );
+    }
+}
+
+impl<P: PedersenConfig> ProductProofTranscriptable for ProductProofIntermediateTranscript<P> {
+    type Affine = sw::Affine<P>;
+    fn add_to_transcript(
+        &self,
+        transcript: &mut Transcript,
+        cx: &Self::Affine,
+        cy: &Self::Affine,
+        cxy: &Self::Affine,
+        ) {
+            ProductProof::make_transcript(
+                transcript,
+                cx, 
+                cy,
+                cxy,
+                &self.alpha,
+                &self.beta,
+                &self.delta,
+            );
+    }
+}
+
+impl<P: PedersenConfig> ProductProof<P> {
+    /// make_intermediate_transcript. This function accepts a set of 
+    /// intermediates and builds an intermediate transcript from those 
+    /// intermediates.
+    /// # Arguments
+    /// * `inter` - the intermediate values to use.
+    pub fn make_intermediate_transcript(
+        inter: ProductProofIntermediate<P>,
+    ) -> ProductProofIntermediateTranscript<P> {
+        ProductProofIntermediateTranscript { 
+            alpha: inter.alpha,
+            beta: inter.beta,
+            delta: inter.delta,
+        }
+    }
+
+    /// make_transcript. This function simply adds `cx`, `cy`, `cxy` and `alpha`
+    /// `beta`, `delta` to the `transcript` object.
+    /// # Arguments
+    /// * `transcript` - the transcript which is modified.
+    /// * `cx` - the commitment to X that is being added to the transcript.
+    /// * `cy` - the commitment to Y that is being added to the transcript.
+    /// * `cxy` - the commitment to XY that is being added to the transcript.
+    /// * `alpha` - the alpha value that is being added to the transcript.
+    /// * `beta` - the beta value that is being added to the transcript.
+    /// * `delta` - the delta value that is being added to the transcript.
+    pub fn make_transcript(
```

Comment:
This is appending to an existing transcript, not making a new one. Maybe `update_transcript`?